src/vs/editor/common/model/textModelSearch.ts

```diff
@@ -89,6 +89,9 @@ export function isMultilineRegexSource(searchString: string): boolean {
 				return true;
 			}
 		}
+		if (chCode === CharCode.LineFeed) {
```

Comment:
The character `\n` (charCode 10) wasn't handled to determine if a regex was multiline. For instance, it was making the test  `AddSelectionToNextFindMatchAction can work with multiline` in `multicursor.test.ts` fails because the search wasn't recognized as multiline.